sql/analyzer/resolve_insert_literals.go

```diff
@@ -0,0 +1,82 @@
+package analyzer
+
+import (
+	"github.com/src-d/go-mysql-server/sql"
+  "github.com/src-d/go-mysql-server/sql/expression"
+	"github.com/src-d/go-mysql-server/sql/plan"
+  errors "gopkg.in/src-d/go-errors.v1"
+
+)
+
+var errWrongNumberOfValues = errors.NewKind("the number of values to insert differ from the expected columns")
+
+func convertIntegerLiteralsInsert(ctx *sql.Context, analyzer *Analyzer, originalNode sql.Node) (sql.Node, error) {
```

Comment:
I think we should let the table implementations handle this. When the literals are parsed, they are converted to the minimum type of int possible, then the table is in charge of converting it.

If we really want to make this transparent for table implementations, then I'd go for doing `column.Type.Convert(theLiteralValue)` inside the `InsertInto` execution method, but I would not add a new rule in the analyzer for this specific case.

WDYT @src-d/data-processing?

Reply:
I think that the conversion should be done by us. If we convert string literals to proper dates, we should do the same here. I'd argue that `int8` is as different from `uint64` as a date is different from a string.

But maybe it makes sense to do it inside InsertInto, not as a new rule. I'll wait for the rest of opinions.

Reply:
Solved in https://github.com/src-d/go-mysql-server/commit/653fdde5b1d0d0a61c1931d187b1909b30571b44.